core/src/main/java/org/lflang/federated/generator/FederateInstance.java

```diff
@@ -220,79 +204,87 @@ public Instantiation getInstantiation() {
   /** Keep a unique list of enabled serializers */
   public HashSet<SupportedSerializers> enabledSerializers = new HashSet<>();
 
+  /** Cached result of analysis of which reactions to exclude from main. */
+  private Set<Reaction> excludeReactions = null;
+
+  /** An error reporter */
+  private final MessageReporter messageReporter;
+
   /**
-   * Return true if the specified EObject should be included in the code generated for this
-   * federate.
+   * Return {@code true} if federate instance inherits the given reactor declaration.
    *
-   * @param object An {@code EObject}
-   * @return True if this federate contains the EObject.
+   * @param declaration The reactor declaration to inherit.
    */
-  public boolean contains(EObject object) {
-    if (object instanceof Action) {
-      return contains((Action) object);
-    } else if (object instanceof Reaction) {
-      return contains((Reaction) object);
-    } else if (object instanceof Timer) {
-      return contains((Timer) object);
-    } else if (object instanceof ReactorDecl) {
-      return contains(this.instantiation, (ReactorDecl) object);
-    } else if (object instanceof Import) {
-      return contains((Import) object);
-    } else if (object instanceof Parameter) {
-      return contains((Parameter) object);
-    } else if (object instanceof StateVar) {
-      return true; // FIXME: Should we disallow state vars at the top level?
-    }
-    throw new UnsupportedOperationException(
-        "EObject class " + object.eClass().getName() + " not supported.");
+  public boolean inherits(ReactorDecl declaration) {
```

Comment:
I'm confused. What does the word "inherits" mean here?  Does it have anything to do with subclassing? Please clarify the documentation and consider renaming the method.

Reply:
I've been struggling with finding the most appropriate name, but I landed on `inherits` because I think that's closest to what happens here.

In my view, there is an original file that defines the federation, and the federate inherits part of the definitions in it. Take for instance the method for determining whether a top-level reaction (i.e., completely one outside of the definition of the federate instance) should be associated with the federate or not; how is it appropriate to call this method `contains` and why doesn't `inherits` cover the meaning of the predicate? Another alternative is `adopts` which avoids the confusion with class inheritance. I just felt that `contains` wasn't good...

The meat of the refactoring, however, was the removal of the public "catch all" overload that used reflection to dispatch to private implementations. This is an anti-pattern because it defers type checking to runtime. Java has overloading, so this was completely unnecessary.

Reply:
Yes, using overloading is a good improvement. But "inherents" I find very confusing. How about "includes"?